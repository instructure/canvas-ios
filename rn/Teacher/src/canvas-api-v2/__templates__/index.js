// @flow
export * from './submission'
export * from './user'
export * from './assignment'
export * from './course'
export * from './group'
export * from './section'
export * from './groupSet'

// QUERIES
export * from '../queries/__templates__/SubmissionList'
export * from '../queries/__templates__/ContextCard'
