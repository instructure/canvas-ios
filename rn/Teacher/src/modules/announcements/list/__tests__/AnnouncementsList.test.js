//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

import React from 'react'
import 'react-native'
import { Refreshed, AnnouncementsList, mapStateToProps } from '../AnnouncementsList'
import app from '../../../app'
import { shallow } from 'enzyme'
import * as template from '../../../../__templates__'

jest
  .mock('Button', () => 'Button')
  .mock('TouchableHighlight', () => 'TouchableHighlight')
  .mock('TouchableOpacity', () => 'TouchableOpacity')
  .mock('../../../../routing/Screen')

describe('AnnouncementsList', () => {
  let props: Props
  beforeEach(() => {
    jest.clearAllMocks()
    app.setCurrentApp('teacher')
    props = {
      context: 'courses',
      contextID: '1',
      refreshing: false,
      refresh: jest.fn(),
      pending: 0,
      error: null,
      navigator: template.navigator(),
      announcements: [
        template.discussion({
          id: '1',
          title: 'Untitled',
          posted_at: '2013-11-14T16:55:00-07:00',
        }),
        template.discussion({
          id: '2',
          title: 'Getting Started in Biology 101',
          posted_at: '2013-10-28T14:16:00-07:00',
        }),
      ],
      refreshAnnouncements: jest.fn(),
      refreshCourse: jest.fn(),
      courseName: 'Im a course',
      courseColor: 'blue',
      permissions: template.discussionPermissions(),
    }
  })

  it('refreshes when there are no announcements', () => {
    const refreshAnnouncements = jest.fn()
    const refreshCourse = jest.fn()
    const refreshProps = {
      announcements: [],
      refreshAnnouncements,
      refreshCourse,
    }

    shallow(<Refreshed {...refreshProps} />)
    expect(refreshAnnouncements).toHaveBeenCalled()
    expect(refreshCourse).toHaveBeenCalledTimes(0)

    refreshProps.context = 'courses'
    shallow(<Refreshed {...refreshProps} />)
    expect(refreshCourse).toHaveBeenCalledTimes(1)
  })

  it('refreshes when there are no permissions', () => {
    const refreshAnnouncements = jest.fn()
    const refreshProps = {
      announcements: [template.discussion({
        id: '1',
        title: 'Untitled',
        posted_at: '2013-11-14T16:55:00-07:00',
      })],
      permissions: {},
      refreshAnnouncements,
    }

    shallow(<Refreshed {...refreshProps} />)
    expect(refreshAnnouncements).toHaveBeenCalled()
  })

  it('refreshes with new props', () => {
    const refreshAnnouncements = jest.fn()
    const refreshProps = {
      announcements: [],
      permissions: {},
      refreshAnnouncements,
    }

    let tree = shallow(<Refreshed {...refreshProps} />)
    tree.instance().refresh()
    tree.setProps(refreshProps)
    expect(refreshAnnouncements).toHaveBeenCalledTimes(2)
  })

  it('renders with no create_announcement permission', () => {
    props.permissions.create_announcement = false
    let tree = shallow(<AnnouncementsList {...props} />)
    expect(tree.find('Screen').prop('rightBarButtons')).toBe(false)
  })

  it('renders while pending', () => {
    props.announcements = []
    props.pending = 1
    let tree = shallow(<AnnouncementsList {...props} />)
    expect(tree.find('ActivityIndicatorView').exists()).toBe(true)
  })

  it('renders while refreshing', () => {
    props.pending = 1
    props.refreshing = true
    let tree = shallow(<AnnouncementsList {...props} />)
    expect(tree.find('ActivityIndicatorView').exists()).toBe(false)
    expect(tree.find('FlatList').prop('refreshing')).toBe(true)
  })

  it('renders empty list', () => {
    props.announcements = []
    let tree = shallow(<AnnouncementsList {...props} />)
    expect(tree.find('FlatList').dive().find('ListEmptyComponent').prop('title'))
      .toBe('There are no announcements to display.')
  })

  it('navigates to new announcement', () => {
    props.contextID = '2'
    let tree = shallow(<AnnouncementsList {...props} />)
    tree.find('Screen').prop('rightBarButtons')[0].action()
    expect(props.navigator.show).toHaveBeenCalledWith(
      '/courses/2/announcements/new',
      { modal: true },
    )
  })

  it('navigates to announcement when row tapped', () => {
    const announcement = template.discussion({ html_url: 'https://canvas.instructure.com/courses/1/discussions/2' })
    props.navigator.show = jest.fn()
    props.announcements = [announcement]
    let tree = shallow(<AnnouncementsList {...props} />)
    tree.find('FlatList').dive().find('Row').simulate('Press')
    expect(props.navigator.show).toHaveBeenCalledWith(announcement.html_url, { modal: false }, {
      isAnnouncement: true,
    })
  })

  it('displays delayed post at date for delayed announcements', () => {
    const announcement = template.discussion({
      delayed_post_at: '3019-10-28T14:16:00-07:00',
      last_reply_at: '2017-10-27T14:16:00-07:00',
    })
    props.announcements = [announcement]
    let tree = shallow(<AnnouncementsList {...props} />)
    let subtitle = tree.find('FlatList').dive().find('[testID="announcements.list.announcement.row-0.subtitle.custom-container"]')
    expect(subtitle.find('[children^="Delayed until:"]').exists()).toBe(true)
    expect(subtitle.find('[children="Oct 28 at 3:16 PM"]').exists()).toBe(true)
  })

  it('displays last reply at date for regular announcements', () => {
    const announcement = template.discussion({
      delayed_post_at: null,
      last_reply_at: '2017-10-27T14:16:00-07:00',
    })
    props.announcements = [announcement]
    let tree = shallow(<AnnouncementsList {...props} />)
    let subtitle = tree.find('FlatList').dive().find('[testID="announcements.list.announcement.row-0.subtitle.custom-container"]')
    expect(subtitle.find('[children="Last post Oct 27 at 3:16 PM"]').exists()).toBe(true)
  })

  it('displays empty post date for announcements with no dates', () => {
    const announcement = template.discussion({
      delayed_post_at: null,
      last_reply_at: null,
    })
    props.announcements = [announcement]
    let tree = shallow(<AnnouncementsList {...props} />)
    let subtitle = tree.find('FlatList').dive().find('[testID="announcements.list.announcement.row-0.subtitle.custom-container"]')
    expect(subtitle.find('[children=""]').exists()).toBe(true)
  })
})

describe('mapStateToProps', () => {
  it('maps course announcement refs to props', () => {
    const one = template.discussion({ id: '1' })
    const two = template.discussion({ id: '2' })
    const three = template.discussion({ id: '3' })
    const state = template.appState({
      entities: {
        courses: {
          '1': {
            announcements: {
              pending: 1,
              error: null,
              refs: ['1', '3'],
            },
            course: {
              name: 'CS 1010',
            },
          },
        },
        discussions: {
          '1': {
            data: one,
          },
          '2': {
            data: two,
          },
          '3': {
            data: three,
          },
        },
      },
    })

    expect(
      mapStateToProps(state, { context: 'courses', contextID: '1' })
    ).toEqual({
      announcements: [one, three],
      pending: 1,
      courseName: 'CS 1010',
      error: null,
    })
  })

  it('maps course announcement refs to props with group context', () => {
    const one = template.discussion({ id: '1' })
    const two = template.discussion({ id: '2' })
    const three = template.discussion({ id: '3' })
    const state = template.appState({
      entities: {
        groups: {
          '1': {
            announcements: {
              pending: 1,
              error: null,
              refs: ['1', '3'],
            },
            group: {
              name: 'CS 1010',
            },
          },
        },
        discussions: {
          '1': {
            data: one,
          },
          '2': {
            data: two,
          },
          '3': {
            data: three,
          },
        },
      },
    })

    expect(
      mapStateToProps(state, { context: 'groups', contextID: '1' })
    ).toEqual({
      announcements: [one, three],
      pending: 1,
      courseName: 'CS 1010',
      error: null,
      permissions: { create_announcement: true, create_discussion_topic: true },
    })
  })

  it('maps empty state', () => {
    const state = template.appState({
      entities: {},
    })
    expect(
      mapStateToProps(state, { context: 'courses', contextID: '1' })
    ).toEqual({
      announcements: [],
      courseName: '',
      courseColor: '',
      pending: 0,
      error: null,
      permissions: {},
    })
  })

  it('maps without course', () => {
    const one = template.discussion({ id: '1' })
    const two = template.discussion({ id: '2' })
    const three = template.discussion({ id: '3' })
    const state = template.appState({
      entities: {
        courses: {
          '1': {
            announcements: {
              pending: 1,
              error: null,
              refs: ['1', '3'],
            },
          },
        },
        discussions: {
          '1': {
            data: one,
          },
          '2': {
            data: two,
          },
          '3': {
            data: three,
          },
        },
      },
    })

    expect(
      mapStateToProps(state, { context: 'courses', contextID: '1' })
    ).toEqual({
      announcements: [one, three],
      pending: 1,
      courseName: '',
      courseColor: '',
      error: null,
    })
  })

  it('puts announcements in order', () => {
    const one = template.discussion({ id: '1', posted_at: '2019-11-15T00:00:00Z' })
    const two = template.discussion({ id: '2', posted_at: '2019-11-16T00:00:00Z' })
    const state = template.appState({
      entities: {
        courses: {
          '1': {
            announcements: {
              pending: 1,
              error: null,
              refs: ['1', '2'],
            },
            course: {
              name: 'CS 1010',
            },
          },
        },
        discussions: {
          '1': {
            data: one,
          },
          '2': {
            data: two,
          },
        },
      },
    })

    expect(
      mapStateToProps(state, { context: 'courses', contextID: '1' })
    ).toMatchObject({
      announcements: [two, one],
    })
  })
})
