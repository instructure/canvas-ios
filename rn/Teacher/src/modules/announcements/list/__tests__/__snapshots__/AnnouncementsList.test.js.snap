// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnnouncementsList refreshes when there are no announcements 1`] = `
<AnnouncementsList
  announcements={Array []}
  refresh={[Function]}
  refreshAnnouncements={
    [MockFunction] {
      "calls": Array [
        Array [
          undefined,
          undefined,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  refreshCourse={[MockFunction]}
  refreshing={false}
/>
`;

exports[`AnnouncementsList refreshes when there are no announcements 2`] = `
<AnnouncementsList
  announcements={Array []}
  context="courses"
  refresh={[Function]}
  refreshAnnouncements={
    [MockFunction] {
      "calls": Array [
        Array [
          undefined,
          undefined,
        ],
        Array [
          "courses",
          undefined,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  refreshCourse={
    [MockFunction] {
      "calls": Array [
        Array [
          undefined,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  refreshing={false}
/>
`;

exports[`AnnouncementsList refreshes when there are no permissions 1`] = `
<AnnouncementsList
  announcements={
    Array [
      Object {
        "allow_rating": false,
        "assignment": Object {
          "anonymize_students": false,
          "assignment_group_id": "1376",
          "automatic_peer_reviews": false,
          "course_id": "987654321",
          "created_at": "2017-03-17T19:15:25Z",
          "description": "a course that can take any beginner to professional in a matter of microseconds.  Harness the power of your favorite languages like PHP and Cold Fusion.",
          "due_at": "2037-06-01T05:59:00Z",
          "grade_group_students_individually": false,
          "grading_type": "points",
          "group_category_id": null,
          "has_overrides": false,
          "html_url": "http://hostname.tld/whatever",
          "id": "123456789",
          "lock_at": "2037-06-01T05:59:59Z",
          "name": "learn to write code",
          "needs_grading_count": 0,
          "only_visible_to_overrides": false,
          "overrides": Array [],
          "peer_reviews": false,
          "points_possible": 20,
          "position": 1,
          "published": true,
          "rubric": null,
          "rubric_assessment": Object {},
          "rubric_settings": null,
          "submission_types": Array [
            "online_text_entry",
            "online_url",
            "media_recording",
            "online_upload",
          ],
          "unlock_at": "2017-03-17T06:00:00Z",
          "unpublishable": false,
          "updated_at": "2017-03-17T19:15:25Z",
        },
        "assignment_id": "123456789",
        "author": Object {
          "avatar_image_url": "http://www.fillmurray.com/100/100",
          "avatar_url": "http://www.fillmurray.com/100/100",
          "display_name": "The Donald",
          "html_url": "",
          "id": "1",
          "short_name": "The Donald",
        },
        "can_unpublish": true,
        "delayed_post_at": null,
        "discussion_subentry_count": 2,
        "discussion_type": "side_comment",
        "html_url": "http://mobiledev.instructure.com/courses/1/discussion_topics",
        "id": "1",
        "is_section_specific": false,
        "last_reply_at": "2016-12-11T04:03:17Z",
        "message": "<p>And why?</p>",
        "only_graders_can_rate": false,
        "permissions": Object {
          "attach": true,
          "delete": true,
          "reply": true,
          "update": true,
        },
        "pinned": false,
        "position": 1,
        "posted_at": "2013-11-14T16:55:00-07:00",
        "published": true,
        "read_state": "read",
        "require_initial_post": false,
        "sort_by_rating": false,
        "subscribed": true,
        "title": "Untitled",
        "unlock_at": null,
        "unread_count": 1,
        "user_can_see_posts": true,
        "user_name": "Ivy Iversen",
      },
    ]
  }
  permissions={Object {}}
  refresh={[Function]}
  refreshAnnouncements={
    [MockFunction] {
      "calls": Array [
        Array [
          undefined,
          undefined,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  refreshing={false}
/>
`;

exports[`AnnouncementsList refreshes with new props 1`] = `
<AnnouncementsList
  announcements={Array []}
  permissions={Object {}}
  refresh={[Function]}
  refreshAnnouncements={
    [MockFunction] {
      "calls": Array [
        Array [
          undefined,
          undefined,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  refreshing={false}
/>
`;

exports[`AnnouncementsList renders 1`] = `
<Screen
  navBarColor="blue"
  navBarStyle="dark"
  onTraitCollectionChange={[Function]}
  rightBarButtons={
    Array [
      Object {
        "accessibilityLabel": "New Announcement",
        "action": [Function],
        "image": Object {
          "testUri": "../../../src/images/Add.png",
        },
        "testID": "announcements.list.addButton",
      },
    ]
  }
  subtitle="Im a course"
  title="Announcements"
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <list>
      <item>
        <TouchableHighlight
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          onPress={[Function]}
          style={
            Array [
              undefined,
              Object {
                "borderBottomColor": "#C7CDD1",
                "borderBottomWidth": 0.5,
              },
            ]
          }
          testID="announcements.list.announcement.row-0"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "flex": 1,
                  "flexDirection": "row",
                  "minHeight": 54,
                  "paddingBottom": 12,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 12,
                },
                Object {
                  "backgroundColor": "#FFFFFF",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                  },
                  Object {
                    "marginLeft": 0,
                  },
                ]
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={0}
                style={
                  Array [
                    Object {
                      "color": "#2D3B45",
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "color": "#2D3B45",
                        "fontWeight": "600",
                      },
                    ],
                  ]
                }
              >
                Untitled
              </Text>
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
                testID="announcements.list.announcement.row-0.subtitle.custom-container"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#2D3B45",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#8B969E",
                        "fontSize": 14,
                        "marginTop": 2,
                      },
                    ]
                  }
                >
                  Last post Dec 10 at 9:03 PM
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-end",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignSelf": "center",
                        "backgroundColor": "transparent",
                        "borderColor": "#C7CDD1",
                        "borderRightWidth": 2,
                        "borderTopWidth": 2,
                        "height": 10,
                        "marginLeft": 7,
                        "width": 10,
                      },
                      Object {
                        "transform": Array [
                          Object {
                            "rotate": "45deg",
                          },
                        ],
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </TouchableHighlight>
      </item>
      <item>
        <TouchableHighlight
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          onPress={[Function]}
          style={
            Array [
              undefined,
              Object {
                "borderBottomColor": "#C7CDD1",
                "borderBottomWidth": 0.5,
              },
            ]
          }
          testID="announcements.list.announcement.row-1"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "flex": 1,
                  "flexDirection": "row",
                  "minHeight": 54,
                  "paddingBottom": 12,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 12,
                },
                Object {
                  "backgroundColor": "#FFFFFF",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                  },
                  Object {
                    "marginLeft": 0,
                  },
                ]
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={0}
                style={
                  Array [
                    Object {
                      "color": "#2D3B45",
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "color": "#2D3B45",
                        "fontWeight": "600",
                      },
                    ],
                  ]
                }
              >
                Getting Started in Biology 101
              </Text>
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
                testID="announcements.list.announcement.row-1.subtitle.custom-container"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#2D3B45",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#8B969E",
                        "fontSize": 14,
                        "marginTop": 2,
                      },
                    ]
                  }
                >
                  Last post Dec 10 at 9:03 PM
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-end",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignSelf": "center",
                        "backgroundColor": "transparent",
                        "borderColor": "#C7CDD1",
                        "borderRightWidth": 2,
                        "borderTopWidth": 2,
                        "height": 10,
                        "marginLeft": 7,
                        "width": 10,
                      },
                      Object {
                        "transform": Array [
                          Object {
                            "rotate": "45deg",
                          },
                        ],
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </TouchableHighlight>
      </item>
    </list>
  </View>
</Screen>
`;

exports[`AnnouncementsList renders empty list 1`] = `
<Screen
  navBarColor="blue"
  navBarStyle="dark"
  onTraitCollectionChange={[Function]}
  rightBarButtons={
    Array [
      Object {
        "accessibilityLabel": "New Announcement",
        "action": [Function],
        "image": Object {
          "testUri": "../../../src/images/Add.png",
        },
        "testID": "announcements.list.addButton",
      },
    ]
  }
  subtitle="Im a course"
  title="Announcements"
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <list>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "height": 100,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
                "fontWeight": "600",
              },
              undefined,
            ]
          }
        >
          There are no announcements to display.
        </Text>
      </View>
    </list>
  </View>
</Screen>
`;

exports[`AnnouncementsList renders while pending 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;

exports[`AnnouncementsList renders while refreshing 1`] = `
<Screen
  navBarColor="blue"
  navBarStyle="dark"
  onTraitCollectionChange={[Function]}
  rightBarButtons={
    Array [
      Object {
        "accessibilityLabel": "New Announcement",
        "action": [Function],
        "image": Object {
          "testUri": "../../../src/images/Add.png",
        },
        "testID": "announcements.list.addButton",
      },
    ]
  }
  subtitle="Im a course"
  title="Announcements"
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <list>
      <item>
        <TouchableHighlight
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          onPress={[Function]}
          style={
            Array [
              undefined,
              Object {
                "borderBottomColor": "#C7CDD1",
                "borderBottomWidth": 0.5,
              },
            ]
          }
          testID="announcements.list.announcement.row-0"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "flex": 1,
                  "flexDirection": "row",
                  "minHeight": 54,
                  "paddingBottom": 12,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 12,
                },
                Object {
                  "backgroundColor": "#FFFFFF",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                  },
                  Object {
                    "marginLeft": 0,
                  },
                ]
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={0}
                style={
                  Array [
                    Object {
                      "color": "#2D3B45",
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "color": "#2D3B45",
                        "fontWeight": "600",
                      },
                    ],
                  ]
                }
              >
                Untitled
              </Text>
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
                testID="announcements.list.announcement.row-0.subtitle.custom-container"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#2D3B45",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#8B969E",
                        "fontSize": 14,
                        "marginTop": 2,
                      },
                    ]
                  }
                >
                  Last post Dec 10 at 9:03 PM
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-end",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignSelf": "center",
                        "backgroundColor": "transparent",
                        "borderColor": "#C7CDD1",
                        "borderRightWidth": 2,
                        "borderTopWidth": 2,
                        "height": 10,
                        "marginLeft": 7,
                        "width": 10,
                      },
                      Object {
                        "transform": Array [
                          Object {
                            "rotate": "45deg",
                          },
                        ],
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </TouchableHighlight>
      </item>
      <item>
        <TouchableHighlight
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          onPress={[Function]}
          style={
            Array [
              undefined,
              Object {
                "borderBottomColor": "#C7CDD1",
                "borderBottomWidth": 0.5,
              },
            ]
          }
          testID="announcements.list.announcement.row-1"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "flex": 1,
                  "flexDirection": "row",
                  "minHeight": 54,
                  "paddingBottom": 12,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 12,
                },
                Object {
                  "backgroundColor": "#FFFFFF",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                  },
                  Object {
                    "marginLeft": 0,
                  },
                ]
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={0}
                style={
                  Array [
                    Object {
                      "color": "#2D3B45",
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "color": "#2D3B45",
                        "fontWeight": "600",
                      },
                    ],
                  ]
                }
              >
                Getting Started in Biology 101
              </Text>
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
                testID="announcements.list.announcement.row-1.subtitle.custom-container"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#2D3B45",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#8B969E",
                        "fontSize": 14,
                        "marginTop": 2,
                      },
                    ]
                  }
                >
                  Last post Dec 10 at 9:03 PM
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-end",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignSelf": "center",
                        "backgroundColor": "transparent",
                        "borderColor": "#C7CDD1",
                        "borderRightWidth": 2,
                        "borderTopWidth": 2,
                        "height": 10,
                        "marginLeft": 7,
                        "width": 10,
                      },
                      Object {
                        "transform": Array [
                          Object {
                            "rotate": "45deg",
                          },
                        ],
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </TouchableHighlight>
      </item>
    </list>
  </View>
</Screen>
`;

exports[`AnnouncementsList renders with no create_announcement permission 1`] = `
<Screen
  navBarColor="blue"
  navBarStyle="dark"
  onTraitCollectionChange={[Function]}
  rightBarButtons={false}
  subtitle="Im a course"
  title="Announcements"
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <list>
      <item>
        <TouchableHighlight
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          onPress={[Function]}
          style={
            Array [
              undefined,
              Object {
                "borderBottomColor": "#C7CDD1",
                "borderBottomWidth": 0.5,
              },
            ]
          }
          testID="announcements.list.announcement.row-0"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "flex": 1,
                  "flexDirection": "row",
                  "minHeight": 54,
                  "paddingBottom": 12,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 12,
                },
                Object {
                  "backgroundColor": "#FFFFFF",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                  },
                  Object {
                    "marginLeft": 0,
                  },
                ]
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={0}
                style={
                  Array [
                    Object {
                      "color": "#2D3B45",
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "color": "#2D3B45",
                        "fontWeight": "600",
                      },
                    ],
                  ]
                }
              >
                Untitled
              </Text>
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
                testID="announcements.list.announcement.row-0.subtitle.custom-container"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#2D3B45",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#8B969E",
                        "fontSize": 14,
                        "marginTop": 2,
                      },
                    ]
                  }
                >
                  Last post Dec 10 at 9:03 PM
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-end",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignSelf": "center",
                        "backgroundColor": "transparent",
                        "borderColor": "#C7CDD1",
                        "borderRightWidth": 2,
                        "borderTopWidth": 2,
                        "height": 10,
                        "marginLeft": 7,
                        "width": 10,
                      },
                      Object {
                        "transform": Array [
                          Object {
                            "rotate": "45deg",
                          },
                        ],
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </TouchableHighlight>
      </item>
      <item>
        <TouchableHighlight
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          onPress={[Function]}
          style={
            Array [
              undefined,
              Object {
                "borderBottomColor": "#C7CDD1",
                "borderBottomWidth": 0.5,
              },
            ]
          }
          testID="announcements.list.announcement.row-1"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "flex": 1,
                  "flexDirection": "row",
                  "minHeight": 54,
                  "paddingBottom": 12,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 12,
                },
                Object {
                  "backgroundColor": "#FFFFFF",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                  },
                  Object {
                    "marginLeft": 0,
                  },
                ]
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={0}
                style={
                  Array [
                    Object {
                      "color": "#2D3B45",
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "color": "#2D3B45",
                        "fontWeight": "600",
                      },
                    ],
                  ]
                }
              >
                Getting Started in Biology 101
              </Text>
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
                testID="announcements.list.announcement.row-1.subtitle.custom-container"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#2D3B45",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#8B969E",
                        "fontSize": 14,
                        "marginTop": 2,
                      },
                    ]
                  }
                >
                  Last post Dec 10 at 9:03 PM
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "flex": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-end",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignSelf": "center",
                        "backgroundColor": "transparent",
                        "borderColor": "#C7CDD1",
                        "borderRightWidth": 2,
                        "borderTopWidth": 2,
                        "height": 10,
                        "marginLeft": 7,
                        "width": 10,
                      },
                      Object {
                        "transform": Array [
                          Object {
                            "rotate": "45deg",
                          },
                        ],
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </TouchableHighlight>
      </item>
    </list>
  </View>
</Screen>
`;
